packages/astro/src/runtime/server/jsx.ts

```diff
@@ -220,11 +220,27 @@ async function renderElement(
 		`<${tag}${spreadAttributes(props)}${markHTMLString(
 			(children == null || children == '') && voidElementNames.test(tag)
 				? `/>`
-				: `>${children == null ? '' : await renderJSX(result, children)}</${tag}>`
+				: `>${
+						children == null ? '' : await renderJSX(result, prerenderElementChildren(tag, children))
+				  }</${tag}>`
 		)}`
 	);
 }
 
+/**
+ * Pre-render the children with the given `tag` information
+ */
+function prerenderElementChildren(tag: string, children: any) {
+	// For content within <style> and <script> tags that are plain strings, e.g. injected
+	// by remark/rehype plugins, or if user explicitly does `<script>{'...'}</script>`,
+	// we mark as HTML string to prevent the content being html-escaped.
```

Comment:
```suggestion
	// For content within <style> and <script> tags that are plain strings, e.g. injected
	// by remark/rehype plugins, or if a user explicitly does `<script>{'...'}</script>`,
	// we mark it as an HTML string to prevent the content from being HTML-escaped.
```